### ISSUE TITLE
Invert QrCode color

### ISSUE THREAD

**Issue by Sata51** [react-native-community/react-native-camera#1566]
Hi,

I'm using [react-native-qrcode-scanner](https://github.com/moaazsidat/react-native-qrcode-scanner)
who is implementing react-native-camera.

The interesting part for me is to be able to invert the color in the barcode detection code. We engrave QrCode with laser on black surface with copper underneath. The result is a QrCode with black background and gold/copper engraving

We are able to read this king of QrCode
![read](https://user-images.githubusercontent.com/17180336/40188664-be7c95be-59fa-11e8-98d2-1a912f23186d.jpg)

But not this kind
![unread](https://user-images.githubusercontent.com/17180336/40188665-bea74688-59fa-11e8-9e6d-c6614c80144d.jpg)

Is there any way provided by react-native-camera to invert color before processing tag ?
Thanks

**Comment by n1ru4l** [comment]
This is currently not possible. If you are still interested in this feature, I encourage you to send a pull request for this feature.

**Comment by jorgeluisrezende** [comment]
There is a way to do it now?

**Comment by n1ru4l** [comment]
@jorgeluisrezende no, but feel free to implement this feature.

**Comment by Sata51** [comment]
I implement this by inverting color on %2 frame to get this work both color

**Comment by jorgeluisrezende** [comment]
@Sata51 how could i do it? can you help me?

**Comment by Sata51** [comment]
@jorgeluisrezende I'm currently not using this package for this purpose. I replace it by react-native-barcode-scanner-google@1.3.0 and change some code in the CameraSource.java like this

```Java

private boolean lastReadInverted = false;

/**
* Called when the camera has a new preview frame.
*/
private class CameraPreviewCallback implements Camera.PreviewCallback {
    @Override
    public void onPreviewFrame(byte[] data, Camera camera) {
        lastReadInverted = !lastReadInverted;
        if (lastReadInverted){
            for (int y = 0; y < data.length; y++) {
                data[y] = (byte) ~data[y];
            }
        }
        mFrameProcessor.setNextFrame(data, camera);
    }
}
```

**Comment by Sjiep** [comment]
@jorgeluisrezende I made a PR (#1896) that should add this feature (inspired by the snippet posted by @Sata51) into the library when using the google vision barcode detector.

**Comment by n1ru4l** [comment]
Implemented in #1896

Please use the master branch or wait until a new release is published on npm!
